spec/integration/rails/persistence_spec.rb

```diff
@@ -149,6 +149,148 @@ def do_put(id)
       end
     end
 
+    describe 'has_one nested relationship' do
+      context 'for new records' do
+        let(:payload) do
+          {
+            data: {
+              type: 'employees',
+              attributes: {
+                first_name: 'Joe',
+                last_name: 'Smith',
+                age: 30
+              },
+              relationships: {
+                salary: {
+                  data: {
+                    :'temp-id' => 'abc123',
+                    type: 'salaries',
+                    method: 'create'
+                  },
+                }
+              }
+            },
+            included: [
+              {
+                :'temp-id' => 'abc123',
+                type: 'salaries',
+                attributes: {
+                  base_rate: 15.00,
+                  overtime_rate: 30.00
+                }
+              }
+            ]
+          }
+        end
+
+        it 'can create' do
+          expect {
+            do_post
+          }.to change { Salary.count }.by(1)
+        end
+      end
+
+      context 'for existing records' do
+        let(:employee) { Employee.create!(first_name: 'Joe') }
+        let(:salary) { Salary.new(base_rate: 15.0, overtime_rate: 30.00) }
+
+        before do
+          employee.salary = salary
+          employee.save
```

Comment:
Let's go with `save!` or `update_attributes!` in tests so the error will be more explicit if we add a validation